src/utils.ts

```diff
@@ -195,6 +195,16 @@ const isAuthorizationCodeGrantConfig = (config: OAuthConfig): config is Authoriz
 const isRefreshGrantConfig = (config: OAuthConfig): config is RefreshGrantConfig =>
   config.grantType === OAuthGrantType.REFRESH_TOKEN_GRANT;
 
+const transformError = (errorToTransform: any) => {
+  let transformedError;
+  try {
+    transformedError = JSON.stringify(errorToTransform).replace(/access_token=[^\s]+/, 'access_token=XXX');
```

Comment:
I have some questions regarding the the regex.

* Do we want to replace everything after the occurence of the string `access_token=[^\s]+`? 
  * How to handle multiline output, do we want to preserve the second line?
* For what reason do we match to whitespace `\s`
* Would it make sense to use another matcher for the access token? @bzums are access tokens strictly base64 encoded?

One example for point 1 and 2 from the above list: <br> `JSON.stringify('access_token=atoken\nmessage 2.n').replace(/access_token=[^\s]+/, 'access_token=XXX');`

Reply:
The whitespace is in there because we want that everything is replaced until the first whitespace, as the declaration of the url ends with a whitespace. Maybe we could instead use a regex that matches a JWT token and exchanges it? That would then of course only exchange access tokens that are jwt tokens.

Reply:
After having a closer look I do not think this is the right place to obfuscate cause we know nothing about the shape of the error. We just assume there will be somewhere `access_token=` in the stringified object and afterwards the token followed by a whitespace but we never know and this can change.

We assume that cause that is how we treat errors in the underlying function: https://github.com/zalando-incubator/authmosphere/blob/092f2c9f95b31d5f2f923f97a06a1ca339cebac0/src/oauth-tooling.ts#L157 and I think there we should do the obfuscation cause we already even log there the url: https://github.com/zalando-incubator/authmosphere/blob/092f2c9f95b31d5f2f923f97a06a1ca339cebac0/src/oauth-tooling.ts#L180 (here we log thee error which also includes the url, so we log even two times...).

Reply:
I spoke with Jan, sounds like a solid idea. He will look into this and update the PR.

Reply:
As discussed briefly, the root error with the access token comes from the fetch library and cannot be obfuscated directly. We'll discuss with @bzums what to do best.